azurelinuxagent/common/cgroup.py

```diff
@@ -13,104 +13,139 @@
 # limitations under the License.
 #
 # Requires Python 2.6+ and Openssl 1.0+
-
+import os
 import re
 
+from azurelinuxagent.common import logger
 from azurelinuxagent.common.exception import CGroupsException
+from azurelinuxagent.common.future import ustr
 from azurelinuxagent.common.osutil import get_osutil
+from azurelinuxagent.common.utils import fileutil
 
-re_user_system_times = re.compile('user (\d+)\nsystem (\d+)\n')
+re_user_system_times = re.compile(r'user (\d+)\nsystem (\d+)\n')
 
 # The metric classes (Cpu, Memory, etc) can all assume that CGroups is enabled, as the CGroupTelemetry
 # class is very careful not to call them if CGroups isn't enabled. Any tests should be disabled if the osutil
 # is_cgroups_support() method returns false.
 
 
 class CGroup(object):
-    def __init__(self):
-        pass
-
-    def _get_cgroup_file(self, hierarchy, file_name):
-        return os.path.join(self.cgroups[hierarchy], file_name)
+    def __init__(self, extension_name, cgroup_path, controller_type):
+        self.name = extension_name
+        self.path = cgroup_path
+        self.controller = controller_type
 
+    def _get_cgroup_file(self, file_name):
+        return os.path.join(self.path, file_name)
 
-    def get_file_contents(self, hierarchy, file_name):
+    def get_file_contents(self, controller, file_name):
         """
-        Retrieve the value of a parameter from a hierarchy.
+        Retrieve the value of a parameter from a controller.
 
-        :param str hierarchy: Name of cgroup metric hierarchy
-        :param str file_name: Name of file within that metric hierarchy
+        :param str controller: Name of cgroup metric controller
+        :param str file_name: Name of file within that metric controller
         :return: Entire contents of the file
         :rtype: str
         """
-        if hierarchy in self.cgroups:
-            parameter_file = self._get_cgroup_file(hierarchy, file_name)
+        if controller == self.controller:
+            parameter_file = self._get_cgroup_file(file_name)
 
             try:
                 return fileutil.read_file(parameter_file)
             except Exception:
-                raise CGroupsException("Could not retrieve cgroup file {0}/{1}".format(hierarchy, file_name))
+                raise CGroupsException("Could not retrieve cgroup file {0}/{1}".format(self.path, file_name))
         else:
-            raise CGroupsException("{0} subsystem not available in cgroup {1}. cgroup paths: {2}".format(
-                hierarchy, self.name, self.cgroups))
+            raise CGroupsException("Incorrect controller: {0} passed for cgroup: {1}. Cgroup path is: {2}".format(
+                controller, self.name, self.path))
 
-    def get_parameter(self, hierarchy, parameter_name):
+    def get_parameter(self, controller, parameter_name):
```

Comment:
it is too since to get confused between get_parameter and get_parameters.

probably this assumption should be verified by the code
> Assumes the parameter is the sole line of the file.

or just ged rid of get_parameter and always use get_parameters